src/Ignitor/SignalsListener/index.ts

```diff
@@ -11,22 +11,36 @@ import { ApplicationContract } from '@ioc:Adonis/Core/Application'
 
 /**
  * Exposes the API to invoke a callback when `SIGTERM` or
- * `SIGINT (pm2 only)` signals are received.
+ * `SIGINT` signals are received.
  */
 export class SignalsListener {
   protected onCloseCallback?: () => Promise<void>
 
+  private isKilling = false
+
   /**
    * Invoke callback and exit process
    */
-  private kill = async function () {
-    try {
-      await this.onCloseCallback()
+  private kill = async () => {
+    if (!this.isKilling) {
+      // First attempt. Try to kill the process gracefully.
+      this.isKilling = true
+      if (process.stderr.isTTY) {
+        // If the process is running in a terminal, display a message to the user.
+        console.error('Gracefully shutting down the process... Press CTRL+C to force it')
```

Comment:
Using `console.error` so the message doesn't end up in the same stream as pino logs and can still be seen by users redirecting stdout to a file.